web/app.js

```diff
@@ -1300,34 +1309,44 @@ var PDFViewerApplication = {
   bindWindowEvents: function pdfViewBindWindowEvents() {
     var eventBus = this.eventBus;
 
+    // Window Resize
+    this._boundEvents.windowResize = (function () {
+      this.dispatch('resize');
+    }).bind(eventBus);
+    // Window Hash Change
+    this._boundEvents.windowHashChange = (function () {
+      this.dispatch('hashchange', {
+        hash: document.location.hash.substring(1),
+      });
+    }).bind(eventBus);
+    // Window Before Print
+    this._boundEvents.windowBeforePrint = (function () {
+      this.dispatch('beforeprint');
+    }).bind(eventBus);
+    // Window After Print
+    this._boundEvents.windowAfterPrint = (function () {
+      this.dispatch('afterprint');
+    }).bind(eventBus);
+    // Window Change
+    this._boundEvents.windowChange = (function (evt) {
```

Comment:
this code shall not be present in non-GENERIC build, can we keep this._boundEvents assignments interleaved with addEventListener